WordPress/Classes/Services/ReaderTopicService.m

```diff
@@ -961,6 +961,19 @@ - (void)mergeMenuTopics:(NSArray *)topics withSuccess:(void (^)(void))success
                         // removing the topic, if it was once followed its not now.
                         topic.following = NO;
                     } else {
+                        // If the user adds a locally saved tag/interest prevent it from being deleted
+                        // while the user is logged out.
+                        if([Feature enabled:FeatureFlagReaderImprovementsPhase2]){
+                            if(!ReaderHelpers.isLoggedIn && [topic isKindOfClass:ReaderTagTopic.class]){
```

Comment:
Not sure I understand. Why do we need to check if not logged in while we're deleting? Isn't it enough to check that `!tagTopic.wasAddedWhileLoggedOut` before deleting?

Reply:
This method is called if the user is logged in or not. If we don't check if the user `isLoggedIn` here, a logged out user will have the followed topics deleted each time.

Basically, what we want to achieve here is to delete followed topics only if the user is now logged in.